smart-contracts/nextjs-escrow/src/lib/anvil-api.ts

```diff
@@ -0,0 +1,234 @@
+const API_ENDPOINT = 'https://preprod.api.ada-anvil.app/v2/services';
+
+const X_API_KEY = process.env.ANVIL_API_KEY;
+
+if (!X_API_KEY) {
+  throw new Error('ANVIL_API_KEY environment variable is not set');
+}
+
+const getHeaders = () => ({
+  'Content-Type': 'application/json',
+  'x-api-key': X_API_KEY,
+});
+
+// Error handling utilities
+const handleApiError = (context: string, error: unknown): string => {
+  console.error(`Error ${context}:`, error);
+  const message = error instanceof Error ? error.message : String(error);
+  return `Failed to ${context}: ${message}`;
+};
+
+// Generic API fetch with error handling
+async function fetchApi<T>(url: string, options: RequestInit, context: string): Promise<T> {
+  try {
+    const response = await fetch(url, options);
+    
```

Comment:
the fetchApi function could take in the endpoint as a param instead of the full url.
```suggestion
// Generic API fetch with error handling
async function fetchApi<T>(endpoint: string, options: RequestInit, context: string): Promise<T> {
  try {
    const response = await fetch(`${API}${endpoint}`, options);
    
```